### ISSUE TITLE
"TypeError: Cannot read properties of undefined (reading 'length')" in math blocks while using rtl plugins

### ISSUE THREAD

**Issue by YonatanRubin** [fenjalien/obsidian-typst#40]
I am not sure if this plugin is the cause for the problem or the rtl plugin, but since this plugin shows the error I assume this one. while using either [dynamic rtl](https://github.com/mwxgaf/obsidian-dynamic-rtl/) or [obsidian-rtl](https://github.com/esm7/obsidian-rtl) with the math block options on.
writing math blocks seem to most of the time show the error "TypeError: Cannot read properties of undefined (reading 'length')" on some blocks, but only when in read mode. this issue persisted both on android and linux. 

in edit mode:
![Screenshot_20240108_081610_Obsidian](https://github.com/fenjalien/obsidian-typst/assets/9606379/c75cb9af-e44c-496d-875a-29cbb2b96d8c)  

in read mode:
![Screenshot_20240108_081556_Obsidian](https://github.com/fenjalien/obsidian-typst/assets/9606379/d4a413f1-6b5f-4aa5-8069-0dace1de0973)

after disabling the rtl plugin:
![Screenshot_20240108_082054_Obsidian](https://github.com/fenjalien/obsidian-typst/assets/9606379/5c258106-329f-423d-beb3-02008f1ba1b0)

**Comment by d-kaue** [comment]
I can't reproduce this with the ObsidianRTL plugin. Are you sure you're getting an error with that one?

The DynamicRTL plugin seems to be deprecated.

**Comment by YonatanRubin** [comment]
yes. I recreated it now in a new vault having only obsidian-rtl and typst-renderer. I used the following file to check if the direction matters, and got it even when the page was set to ltr (which is weird because it is the first time it behaved like this that I have seen)
```markdown
test for only ltr in my code.
a few lines
inline $f_x (t) = integral_0^oo e^(k/2) d t$
block :
$$f_x (t) = integral_0^oo e^(k/2) d t$$
multiline block:
$$f_x (t) = integral_0^oo e^(k/2) d t\ 
F'_x (t)=f_x (t)
$$
```

**Comment by d-kaue** [comment]
with your example I was able to reproduce it.

e.g. this is OK.
```md
Lorem $Alpha$ ipsum.

Lorem $Beta$ ipsum.
```

but this will get an error.
```md
Lorem $Alpha$ ipsum.
Lorem $Beta$ ipsum.
```

Obsidian recognises it as a single `<p>` element and insert a `<br>` line break. In this case obsidian-rtl creates a `<div>`.

with obsidian-rtl **enabled**
![with obsidian-rtl](https://github.com/fenjalien/obsidian-typst/assets/25023130/eb7d1dd1-18d4-4e53-8fe9-341883ffa9bd)

with obsidian-rtl **disabled**
![without obsidian-rtl](https://github.com/fenjalien/obsidian-typst/assets/25023130/a4ffd116-aaab-48a0-9e5a-9ff1abb4b813)


obsidian-rtl separates each element inside the `<p>` element that contains a `<br>` line break and creates a `<div>` in each element.
the typst compiler is probably losing the reference of the element it is rendering.

To be honest, I don't know anything about Left-to-Right. The content inside each inline maths have to change position?

**Comment by fenjalien** [comment]
I can reproduce this as well, it looks like at some point the custom HTML element that renders the Typst source code gets reset or recreated. The error is from the source string being `undefined` but still being read as a string. This shouldn't happen as it gets given the source code so I'm not entirely sure whats happening.

**Comment by YonatanRubin** [comment]
what do you mean by change position? if you mean change position in relation to the normal text then probably (since a line with n chars before the math block should position it either n pixels to the correct alignment). if you mean compared to itself then no, the math should be in the same direction orientation and position.

**Comment by d-kaue** [comment]
> I can reproduce this as well, it looks like at some point the custom HTML element that renders the Typst source code gets reset or recreated. The error is from the source string being `undefined` but still being read as a string. This shouldn't happen as it gets given the source code so I'm not entirely sure whats happening.

I've been debugging for a while, and it seems to be a problem with the `CustomElements` lifecycle.
obsidian-rtl modifies before draw, but after obsidian-typst creating the element.

In the `TypstRenderElement` class the `draw` method only has an idea of the `fontSize` and `this.size` because it is defined inside it, everything else that is passed in the `.compile` is `undefined`.

**Comment by YonatanRubin** [comment]
obsidian just added better rtl support, the obsidian rtl plugin no longer gives this error!
